python/tank/authentication/sso_saml2/core/sso_saml2_core.py

```diff
@@ -222,54 +315,97 @@ def acceptNavigationRequest(self, frame, request, n_type):  # noqa
         self._dialog.setWindowTitle(window_title)
         self._dialog.finished.connect(self.on_dialog_closed)
 
-        self._view = QtWebKit.QWebView(self._dialog)
-        self._view.setPage(TKWebPage(self._dialog))
-        self._view.page().networkAccessManager().authenticationRequired.connect(
-            self.on_authentication_required
-        )
-        self._view.loadFinished.connect(self.on_load_finished)
-
-        # We want to inject custom JavaScript code before any code is
-        # executed in the loaded web pages. This is to polyfill any
-        # missing functionality.
-        frame = self._view.page().currentFrame()
-        frame.javaScriptWindowObjectCleared.connect(self._polyfill)
+        # This is to ensure that we can resize the window nicely, and that the
+        # WebView will follow.
+        self._layout = QtGui.QVBoxLayout(self._dialog)
+        self._layout.setSpacing(0)
+        self._layout.setContentsMargins(0, 0, 0, 0)
 
-        # Purposely disable the 'Reload' contextual menu, as it should not be
-        # used for SSO. Reloading the page confuses the server.
-        self._view.page().action(QtWebKit.QWebPage.Reload).setVisible(False)
+        if QtWebKit:
+            # Disable SSL validation, to allow self-signed certs for local installs.
```

Comment:
This indeed used to be behind a developer flag. But to allow local sites that still use self-signed certs to make use of the Unified Login Flow, we need to relax the cert validation.

This is not strictly needed for this PR, but it will be for when I enable the Unified Login Flow by default.

And if you look at the `TKWebPageQt5` class, we also relax the cert check in `certificateError()`.

So either I put both of these in now, or I put them both in a later PR. Let me know which one you prefer.  (Or alternatively I could put both behind the developer flag, but this would only be temporary until my next PR when enabling ULF).

Reply:
Wouldn't ULF be enabled from desktop startup and not tk-core?

Reply:
You are correct, the ULF flow would be initiated from the Desktop, via something set in the tk-core's authentication code (like a semi-global flag). But that knowledge would not be exposed to the sso_saml2 library, which does not know that it is used for every login.